README.md

```diff
@@ -1,45 +1,617 @@
+}
 ```
 
-## Samples
+Request parameters:
+- `storeId` - the ID of your assigned store
+- `paymentReference` - a unique reference for the payment, sent from the client and returned by the server
+- `amount` - the amount to request for the online payment, formatted as a 2.dp decimal number, such as `1.00`
+- `currency` - the currency for this payment in 3 character ISO format, such as `GBP`
+- `orderId` - a unique order ID for this payment, sent from the client and returned by the server
+- `returnUrl` - where the payment form will redirect to after the payment has completed, the return url will have additional query parameters appended to indicate the status of the payment request.
+- `expiry` - (optional) setting expiry settings
+  - `showTimer` - (optional) flag to show countdown timer in payment page
+  - `expiringSeconds` - (optional) how long the payment form will wait until the payment expires and the page will redirect to the return url
+
+
+#### PayResponse : Payment Response
+
+````java
+public enum ErrorType {
+  UNKNOWN("UNKNOWN"),
+  DUPLICATE_ATTEMPT("DUPLICATE_ATTEMPT"),
+  INVALID_REQUEST("INVALID_REQUEST"),
+  CANCELLED("CANCELLED")
+}
+
+public class Error {
+  private ErrorType type;       //Enum for the error type
+  private String message;       //Error message
+}
+
+public class PaymentInitiationResponse {
+    private Response response;
+    private Error error;
+}
+
+public class Response {
+  private String paymentId;     // Unique identifier created by Kody
+  private String paymentUrl;    // The URL to send to the user from your application 
+}
+````
+
+#### Java Demo
+````java
+import common.PaymentClient;
+import com.kodypay.grpc.pay.v1.*;
+
+PaymentClient paymentClient = new PaymentClient(HOSTNAME, APIKEY);
+
+String storeId = "UUID of assigned store";
+String orderId = "a unique reference for the payment";
+String paymentReference = "a unique reference for the payment";
+String amount = "1.00";
+String currency = "GBP";
+String returnUrl = "returnUrl";
+ExpirySettings expirySettings = PaymentInitiationRequest.ExpirySettings.newBuilder()
+        .setShowTimer(true)
+        .setExpiringSeconds(1800)
+        .build();
+
+PaymentInitiationRequest paymentInitiationRequest = PaymentInitiationRequest.newBuilder()
+        .setStoreId(storeId)
+        .setPaymentReference(paymentReference)
+        .setAmount(amount)
+        .setCurrency(currency)
+        .setOrderId(orderId)
+        .setReturnUrl(returnUrl)
+        .setExpiry(expirySettings)
+        .build();
+
+PaymentInitiationResponse response = paymentClient.sendOnlinePayment(paymentInitiationRequest);
+````
+
+### Get Online Payments
+
+The payment details request requires the following parameters:
+- `storeId` - the ID of your assigned store
+- `pageCursor` - set pagination settings
+  - `page` - set an offset of the results. Default value is 0
+  - `pageSize` - the number of results to be returned per page. Default value is 0
```

Comment:
If both defaults are zero, what is returned?
All pages starting from the first page?

Reply:
The default `pageSize` should be `1` I think

Reply:
From what I read LIMIT 0 quickly returns an empty set. 
OFFSET 0 is the same as not having an offset